libraries/mbed/targets/hal/TARGET_Freescale/TARGET_K20XX/i2c_api.c

```diff
@@ -56,8 +45,26 @@ void i2c_init(i2c_t *obj, PinName sda, PinName scl) {
     obj->i2c = (I2C_Type*)pinmap_merge(i2c_sda, i2c_scl);
     MBED_ASSERT((int)obj->i2c != NC);
 
+    
+    
+#if defined(TARGET_K20DX256)
+    switch ((int)obj->i2c) {
+        case I2C_0: SIM->SCGC4 |= 1 << 6; break;
+        case I2C_1: SIM->SCGC4 |= 1 << 7; break;
+
+       		 /*
+        	// enable power
```

Comment:
They can be deleted. The old code (which originally started on the KL25z and everyone just copies) enables both the I2C peripheral and the relevant output port. The new code only the relevant I2C peripheral. This because enabling the output port is incomplete (they can be connected to multiple ports), and additionally it is done already by the pin_function code, which is called by the pinmap code. 

So thats why it was changed, and old one can be deleted. For the Teensy it should probably also be changed to use the bit names from register definition file. I am wondering also if we should stick to the current #ifdef or use the TARGET_TEENSY3_1 define. This one was kept from a name change, but should still be included in the build. And in principle it might be nicer to use the MCU name then the board name.